### PR TITLE
[13.0][MIG] hr_timesheet_activity_begin_end: Migration to 13.0

### PR DESCRIPTION
**Pull request by davejames**

### CODE REVIEW

**Review by alexey-pelykh — APPROVED** [review]

**Review by astirpe — CHANGES_REQUESTED** [review]
Please cherry-pick https://github.com/OCA/timesheet/pull/307/commits/bfa417eeecba68c129044baeeda0b0104bb3c52b from https://github.com/OCA/timesheet/pull/307

**Comment by davejames** [comment]
@astirpe I have re-done this migration and your changes are now included

**Comment by davejames** [comment]
@pedrobaeza any chance you could review this?

**Comment by pedrobaeza** [comment]
@astirpe should approve your latest changes.

File: hr_timesheet_activity_begin_end/models/account_analytic_line.py [hunk 1]
```diff
@@ -15,9 +15,9 @@ class AccountAnalyticLine(models.Model):
     time_start = fields.Float(string="Begin Hour")
     time_stop = fields.Float(string="End Hour")
 
-    @api.one
     @api.constrains("time_start", "time_stop", "unit_amount")
     def _check_time_start_stop(self):
+        self.ensure_one()
```
Comment (astirpe) [inline hr_timesheet_activity_begin_end/models/account_analytic_line.py]: Here self could be a recordset, not necessarily a singleton.
This constraint should be rewritten this way:

```python
    @api.constrains("time_start", "time_stop", "unit_amount")
    def _check_time_start_stop(self):
        for line in self:
            value_to_html = self.env["ir.qweb.field.float_time"].value_to_html
            start = timedelta(hours=line.time_start)
            stop = timedelta(hours=line.time_stop)
            ....
            etc..
            ....
```

File: hr_timesheet_activity_begin_end/__manifest__.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 {
     "name": "Timesheet Activities - Begin/End Hours",
-    "version": "12.0.1.1.0",
+    "version": "13.0.1.0.0",
     "author": "Camptocamp, " "Tecnativa, " "Odoo Community Association (OCA)",
```
Comment (astirpe) [inline hr_timesheet_activity_begin_end/__manifest__.py]:
```suggestion
    "author": "Camptocamp, Tecnativa, Odoo Community Association (OCA)",
```

**Review by astirpe — CHANGES_REQUESTED** [review]

File: hr_timesheet_activity_begin_end/models/account_analytic_line.py (hunk 1 above)
Comment (astirpe) [reply hr_timesheet_activity_begin_end/models/account_analytic_line.py]: You forgot the loop on self

File: hr_timesheet_activity_begin_end/models/account_analytic_line.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            if others:
+                message = _("Lines can't overlap:\n")
+                message += "\n".join(
+                    [
+                        "%s - %s"
+                        % (
+                            value_to_html(line.time_start, None),
+                            value_to_html(line.time_stop, None),
+                        )
+                        for other in (line + others).sorted(lambda l: l.time_start)
```
Comment (astirpe) [inline hr_timesheet_activity_begin_end/models/account_analytic_line.py:63]: Now variable `other` is not used. Please check

**Review by astirpe — APPROVED** [review]
Code LGTM. Not tested.

**Comment by davejames** [comment]
> Code LGTM. Not tested.

thanks for your review and patience :)

**Pull request merged**